src/cloud/subdeployment/index.js

```diff
@@ -0,0 +1,216 @@
+import fs from 'fs';
+import path from 'path';
+import yaml from 'js-yaml';
+import { get } from 'lodash';
+import { green, gray, yellow } from 'kleur';
+import { exit } from '../../util/exit';
+import { prompt } from '../../util/prompt';
+import { assertBedrockRoot } from '../../util/dir';
+import { validateSimpleName } from '../../util/validation';
+import { readServiceYaml, writeServiceYaml, getServiceFilePath } from '../utils';
+import { exec, execSyncInherit } from '../../util/shell';
+import { checkService } from '../utils';
+import { checkConfig } from '../authorize';
+
+const TAG_REG = /(:\w+)?$/;
+const DOMAIN_REG = /^https?:\/\/([^/]+)/;
+const INGRESS_REQUIRED_VERSION = 'networking.k8s.io/v1';
+
+export async function create(options) {
+  await assertBedrockRoot();
+  await checkConfig(options);
+  await checkService(options);
+
+  if (options.name) {
+    const err = validateSimpleName(options.name);
+    if (typeof err === 'string') {
+      exit(err);
+    }
+  } else {
+    options.name = await prompt({
+      type: 'text',
+      message: 'Subdeployment simple name (one word or hyphenated):',
+      validate: validateSimpleName,
+    });
+  }
+
+  const { name, environment, service } = options;
+  const deployFile = `${service}-deployment.yml`;
+  const deployment = readServiceYaml(environment, deployFile);
+  updateDeployment(deployment, options);
+
+  if (!options.domain) {
+    exit(`Could not derive the domain from ${deployFile}.`);
+  }
+  if (!options.port) {
+    exit(`Could not derive the service port from ${deployFile}.`);
+  }
+
+  writeServiceYaml(environment, `${name}-${deployFile}`, deployment);
+  await patchIngress(options);
```

Comment:
No need to await patchIngress, unless patchIngress needs to change into an async function.

Reply:
ah good catch... actually `patchIngress` should be async and await the patch update, although this will return fairly quick and the actual ingress sync will still take time